### PR TITLE
task(Build): Update packages, minor refactor for build warnings, target only netstandard2.0 and net472 

### PR DESCRIPTION
**Pull request by jafin**
* task (Packages): Update packages Net.Test.Sdk > 16.5 , PublicApiGenerator => 10.0.2  TestStack.Conventions => 4.0.2
* task (Build): Should mark as serializable with attribute, remove unused args from Main.
* Fix(GitVersion): GitVersionConfig.yaml is deprecated, replaced with gitversion.yml
* target netstandard2.0 and net472, removed netstandard1.3 and net451

### CODE REVIEW

**Comment by jafin** [comment]
This may or may not be wanted, I was just doing some tidy up work locally, so I thought I'd push it up as a PR anyhow.   You may want to keep 451 or netstd 1.3.  
Let me know can update if required.

File: src/Shouldly.Tests/ConventionTests/ApprovePublicApi.cs [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
         [IgnoreOnAppVeyorLinuxFact]
         public void ShouldlyApi()
         {
-            var publicApi = PublicApiGenerator.ApiGenerator.GeneratePublicApi(typeof(Should).Assembly, null,false);
+            var options = new ApiGeneratorOptions()
+            {
+                IncludeTypes = null,
+                IncludeAssemblyAttributes = false
+            };
+            var publicApi = PublicApiGenerator.ApiGenerator.GeneratePublicApi(typeof(Should).Assembly, options);
```
Comment (sungam3r) [inline src/Shouldly.Tests/ConventionTests/ApprovePublicApi.cs:15]: `typeof(Should).Assembly.GeneratePublicApi(options)`

**Comment by sungam3r** [comment]
Maybe PublicApiGenerator changes should go in separate PR (without TFM changes).

**Comment by jafin** [comment]
@sungam3r sure can separate into separate PR's

**Comment by jafin** [comment]
@sungam3r  split into:
#612 Fix(GitVersion): GitVersionConfig.yaml is deprecated, replaced with gitversion.yml
#613 task (PublicApiGenerator): Update PublicApiGenerator to 10.0.2
#614 task (TestStack): Update TestStack => 4.0.2, update Net.test.Sdk =>16.5 Fix spirious d in DefineConstants

**Pull request closed without merging**